Add home items to cart, display items on shopping cart

lehone-hp opened the pull request:
@muarachmann code review.
- Displayed the cart items on the shopping-cart icon at the top right hand corner
- I used used onclick() on items on the home page when adding to cart. Please review this approach and if you think it is good I will implement on the other pages and update this PR.

muarachmann commented on resources/views/layouts/app.blade.php:
```diff
@@ -314,6 +322,104 @@
+                    console.log(data.msg);
+                    alert(data.msg);
+                    $(".cartCount").html(data.cart_count);
+                    updateCartList();
```
Works perfectly ok!

muarachmann commented on resources/views/layouts/app.blade.php:
```diff
@@ -314,6 +322,104 @@
+        });
+
+        //TODO make a request to the server to get all items on the cart
+        //TODO Update the cart_product_list
```
I think its is okay.. Another way to load the cart in all views is to use ( a variable in laravel that can be shared amongst all views). Check on that or we could still go this way which is stressful IMO. Anyway you choose aside that all is good to go

lehone-hp replied:
I don't understand what you mean by using a variable that can be shared in all the views

lehone-hp commented:
I have updated the PR
- Added a Country model with migration to insert all countries in the table.
- Fill checkout form with user's info i.e. if the user is logged in but leave the fields empty otherwise. 
Review so we can move ahead.

muarachmann commented:
ok @lehone-hp will test

muarachmann commented:
Tested and it works fine. Great Job. Just left the remove cart item and condition of if in stock or not. When done i will make a PR for a better toast that alert box dialog

muarachmann commented:
@lehone-hp you can go ahead and merge this 👍
